libraries/Microsoft.Bot.Builder.Dialogs/DialogContext.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Dialogs
+{
+    public class DialogContext
+    {
+        private object _finalResult;
+
+        public DialogSet Dialogs { get; set; }
+        public ITurnContext Context { get; set; }
+        public Stack<DialogInstance> Stack { get; set; }
+
+        /// <summary>
+        /// Creates a new DialogContext instance.
+        /// </summary>
+        /// <param name="dialogs">Parent dialog set.</param>
+        /// <param name="context">Context for the current turn of conversation with the user.</param>
+        /// <param name="stack">Current dialog stack.</param>
+        public DialogContext(DialogSet dialogs, ITurnContext context, Stack<DialogInstance> stack)
+        {
+            Dialogs = dialogs;
+            Context = context;
+            Stack = stack;
+        }
+
+        /// <summary>
+        /// Returns the cached instance of the active dialog on the top of the stack or `undefined` if the stack is empty.
+        /// </summary>
+        public DialogInstance Instance
```

Comment:
This method works like no instance method I've *EVER* seen. 

Please revisit.

Reply:
**By Design**

Reply:
we can work on the design - this is exactly consistent with the JavaScript
(my opinion: its not beautiful but its not that bad)